vsintegration/src/FSharp.Editor/CodeFix/AddMissingRecToMutuallyRecFunctions.fs

```diff
@@ -20,11 +20,9 @@ open Microsoft.CodeAnalysis.CodeActions
 type internal FSharpAddMissingRecToMutuallyRecFunctionsCodeFixProvider
     [<ImportingConstructor>]
     (
-        projectInfoManager: FSharpProjectOptionsManager
     ) =
     inherit CodeFixProvider()
 
-    static let userOpName = "AddMissingRecToMutuallyRecFunctions"
```

Comment:
Shouldn't we keep these for causality tracking?  I mean, I know our synchonrous stacks are getting a lot better.  But I suspect for logging etc. we still want to track operations back via a propagated string like this - otherwise it's really hard to tell what operation is triggering actions, especially if anything along the way is asynchronous.

Concretely this would mean `GetFSharpCompilationDefinesAsync() ` taking and propagating a `userOpName`

Reply:
I did some debug testing and the stack traces are even better with this PR because of the caching that I am doing with the Roslyn `Project` and `FSharpProjectOptions` - using the conditional weak table. I really do not think we need to add the strings back in now. I'm a bit biased against the strings; I really just want to reduce the amount of code that we have to do in order to do simple tasks.

![good_stack1](https://user-images.githubusercontent.com/1278959/123148851-7e1a8100-d415-11eb-8fa8-2d8001ed36d1.png)


Combined with this PR: https://github.com/dotnet/fsharp/pull/11715 - stacks are even better:

![good_stack2](https://user-images.githubusercontent.com/1278959/123148865-82469e80-d415-11eb-9f84-d564801adb41.png)

While the caching isn't perfect; the first time it has to compute new project options the stacks will not be as good - but from what I can tell, the call Roslyn makes that requires new options is brace matching - then all subsequent calls after that including semantic classification and diagnostics will hit the cache and the stacks will be maintained.

Reply:
I added back in the strings. If we find a better way to handle it; it will be a separate PR. Sorry for trying to include this change in the current PR. The goal was to make things less complicated.